components/profile_popover/profile_popover.tsx

```diff
@@ -234,6 +236,13 @@ ProfilePopoverState
             },
         ));
     }
+
+    componentDidUpdate(prevProps: ProfilePopoverProps) {
+        if (this.props.openModalCount !== prevProps.openModalCount) {
+            this.props.hide?.();
+        }
```

Comment:
Does this need to know the modal *count* or does it just need to know that *any* modal is open (using the `isAnyModalOpen` selector)?

Reply:
I thought of the edge case that we'll need that info on a modal. In this case this way will monitor for change (closing the modal) and dismiss it as well. Maybe we could opt in using the selector if there isn't a possible case at the moment or near future. What do you think?

Reply:
I don't understand.

Reply:
This is a popover showing information about a user or a group. This is something that could potentially exist in a popup. In that case, there will be a popup open, and by counting modals it would work in that case. However this could be an edge case.

Reply:
For some reason I'm completely not understanding.

Reply:
Maybe because it's not a valid reason 😅

Let's take the insights page as an example. At some point, we decide that we don't want onClick to navigate to DM with the user but to show user information using the profile popover.

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/684680/209404864-6d091022-d095-460b-8de4-9a1787df47f8.png">

In this case, there is one modal, so the `isAnyModalOpen` will return `true`. Pressing `cmd+K` or `cmd+/` will trigger another modal:

https://user-images.githubusercontent.com/684680/209405676-f81faa31-0c0a-4524-a84b-49ac05bb882d.mp4

So in the current implementation, it will go from 1 modal to 2 modals so it will hide the popover. But watching the `isAnyModalOpen` it will remain `true` so the popover will remain visible and have the same issue as right now.

I'm probably overthinking it, and there isn't a current issue with that I could find. It's just seems to me a bit more future-proof to me, but I don't have a strong opinion I could probably reuse the existing solution, which would be pretty solid for 99.9% of the cases

Reply:
i am with @mkraft on this, and i have gone through the above comments, i do think we are probably over engineering, but isAnyModalOpen can work here :P

Reply:
Thanks, both for the suggestion. I was definitely overthinking this situation.